### DISCUSSION TITLE
Pruning trial non promising trials logic

### DISCUSSION THREAD

**Discussion by eduardcmp** [optuna/optuna#4253] [Q&A]
I'm hypertuning a XGBmodel and it seems that when I have a wide range of parameters and in the early trials high values for the learning rate and max depth parameters are used then most of the next trials are pruned.

I wonder if when deciding wether to prune an unpromising trail it's considered the complexity of the underlying model and the learning rate. Because if it only considers the value of the objective function at iteration i, the obviously there will be a bias towards higher learning rates and more complex base models.

Thanks in advance.

**Comment by amylase** [comment]
I think you are right and this is an open question.
[The hyperband paper](https://jmlr.org/papers/volume18/16-558/16-558.pdf) mentions this problem as an open challenge in the conclusion.

> Adjusting for different convergence rates. 

> A second open challenge involves generalizing the ideas behind Hyperband to settings where configurations have drastically
differing convergence rates. Configurations can have different convergence rates if they
have hyperparameters that impact convergence (e.g., learning rate decay for SGD or neural
networks with differing numbers of layers or hidden units), and/or if they correspond to
different model families (e.g., deep networks versus decision trees). The core issue arises
when configurations with drastically slower convergence rates ultimately result in better
models. To address these issues, we should be able to adjust the resources allocated to each
configuration so that a fair comparison can be made at the time of elimination.

**Reply by eduardcmp** [reply]
Thanks for your reply. That's exactly the problem I identified. I'll assume it's not solved yet in the package. I believe it's a really important problem unless the param space we're investigating doesn't have important differences in terms of convergence speed.
